k-distribution/tests/regression-new/equals-pattern/test.k

```diff
@@ -3,6 +3,8 @@
 module TEST-SYNTAX
   imports INT
   imports K-EQUAL
+  imports BOOL
+  imports BASIC-K
```

Comment:
I'm a little concerned that having a long list of modules that most semantics would need to import would lead to semantics just importing `DOMAINS`. I'm not sure if that would negate the modularity benefits of having them private in the first place?

Could we merge `K-EQUAL` into `BOOL`?
Could we have `BASIC-K` imported by default?

Reply:
BASIC-K actually is imported everywhere by default except the program parser, where we deliberately import nothing implicitly. The import exists here because we actually want to include it in the program parser for this definition.

We could maybe make `BOOL` import K-EQUAL publicly though. What do others think about this?